setup.py

```diff
@@ -17,10 +18,33 @@
 # You should have received a copy of the GNU General Public License
 # along with ShinySDR.  If not, see <http://www.gnu.org/licenses/>.
 
+# This setup.py is Based on the work from:
+# https://github.com/kennethreitz/setup.py/blob/master/setup.py
+
+import os
+import urllib
+import subprocess
+
 from setuptools import find_packages, setup
 
+ASSETS = {
+    'http://requirejs.org/docs/release/2.1.22/comments/require.js': 'shinysdr/deps/require.js',
+    'https://raw.githubusercontent.com/requirejs/text/646db27aaf2236cea92ac4107f32cbe5ae7a8d3a/text.js': 'shinysdr/deps/text.js'
+}
+
+SETUP_DIR = os.path.abspath(os.path.dirname(__file__))
+
+try:
+    subprocess.call(['git', 'submodule', 'update', '--init'])
```

Comment:
This is problematic because it means there are side effects just for running, say, `setup.py --help`. It should only do anything if an actual operation is being run (but there should still be a way to fetch deps without a build, for run-from-source-tree operation as documented in https://shinysdr.switchb.org/manual/installation).

Reply:
this is now a set of sub commands 'init' for git submodule init and 'retrieve' for downloading assets like require.js.

Reply:
Also added to 'build' so only setup.py build is needed, but 'git_init' and 'retrieve_assets' can still be used independently.

Reply:
There should be a single command which run-from-source users can use to complete both the HTTP and Git fetches. It could be called "fetch_deps". (Then as well as being documented in `installation.html`, it can be used in the travis configuration in place of `build`, to save some steps and avoid being misleading since the build result isn't being used.)

(I say "fetch_deps", not "fetch_js_deps", because that name was slightly wrong since the addition of the APRS symbols library.)

Reply:
OK, this all looks good, except — it looks like the part where "build" implicitly does "fetch_deps" now? My understanding of the point of this PR is that running setup.py using standard package build/installation tools/steps should fetch deps.

Reply:
On second thought I'll take care of this on a branch of my own.

Reply:
Sorry that was a misunderstanding on my part of your original request for fetch_deps. I can easily add back the automatic build pull of fetch_deps if you want.